### PR TITLE
libzip: bump version of dependencies + use cmake_find_package generator

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libzip/all**

Fixes https://github.com/conan-io/conan-center-index/issues/5108

@5783mgordon Please test

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libzip/all/conanfile.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class LibZipConan(ConanFile):
     license = "BSD-3-Clause"
     topics = ("conan", "zip", "libzip", "zip-archives", "zip-editing")
     exports_sources = ["CMakeLists.txt"]
-    generators = "cmake"
+    generators = "cmake", "cmake_find_package"
```
Comment (SpaceIm) [inline recipes/libzip/all/conanfile.py:14]: Seems to be ok for most dependencies, except zstd. Upstream relies on a custom FindZstd.cmake, and cmake_find_package will generate Findzstd.cmake

Comment (madebr) [reply recipes/libzip/all/conanfile.py:14]: Maybe remove conan's `FindZstd.cmake` file to give us more assurance?

Comment (SpaceIm) [reply recipes/libzip/all/conanfile.py:14]: Not enough, find_package(Zstd) won't find Findzstd.cmake (uppercase vs lowercase).

Moreover, their custom FindZstd.cmake defines a target `Zstd::Zstd`, so upstream CMakeLists links to this target, which doesn't exist in conan cmake_find_package.

Comment (madebr) [reply recipes/libzip/all/conanfile.py:14]: I meant removing conan's `Findzstd.cmake`.
Or what do you suggest?
Or is the current state ok?

Comment (SpaceIm) [reply recipes/libzip/all/conanfile.py:14]: Didn't you add `cmake_find_package` to not use custom upstream or built-in Find files, or I missed something?

Comment (madebr) [reply recipes/libzip/all/conanfile.py:14]: I added it for #5108.
He's seeing missing dl/pthread symbols, used by openssl.

Comment (SpaceIm) [reply recipes/libzip/all/conanfile.py:14]: Ok I understand, but zstd injection seems to be fragile.

In upstream CMakeLists, I would replace `find_package(Zstd)` by:

```cmake
find_package(zstd REQUIRED)
set(Zstd_FOUND 1)
add_library(Zstd::Zstd INTERFACE IMPORTED)
set_property(TARGET Zstd::Zstd PROPERTY INTERFACE_LINK_LIBRARIES zstd::zstd)
```

Comment (madebr) [reply recipes/libzip/all/conanfile.py:14]: I'll open a pr at https://github.com/nih-at changing their casing + name of targets.
Then no more patches will be required for future releases.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by theirix — APPROVED** [review]

**Pull request merged**